build-and-test-with-yarn/action.yaml

```
@@ -24,13 +24,25 @@ inputs:
     description: Comma-separated paths to directories containing main source files.
     default: src/
     required: false
-  sonar-test-exclusions:
-    description: One or more path matching patterns defining files to be excluded from analysis, among those found below the directories specified in `sonar.tests`.
-    default: test/**
+  sonar-inclusions:
+    description: One or more path matching patterns defining files to be included in analysis, among those found below directories specified in `sonar.sources`.
+    default: "**/*.ts"
+    required: false
+  sonar-exclusions:
+    description: One or more path matching patterns defining files to be excluded from analysis, among those found below the directories specified in `sonar.sources`.
+    default: "**/*.spec.ts"
     required: false
   sonar-tests:
     description: Holds a comma-delimited list of directories (not a path matching pattern) relative to the current working directory.
-    default: test/
+    default: src/
+    required: false
+  sonar-test-inclusions:
+    description: One or more path matching patterns defining files to be included in analysis, among those found below directories specified in `sonar.tests`. Applied after `sonar.test.exclusions` and therefore overriding it.
+    default: "**/*.spec.ts"
```

Dont we use test.ts?
```suggestion
    default: "**/*.test.ts"
```

---

The default for nest.js is `spec` and I remember this form other node.js apps.

---

As agreed during the Daily I have amended this to match `test.ts`